src/utils/coroutine-utils.ts

```diff
@@ -51,17 +46,16 @@ export class JobRunner<T> {
   }
 
   tick() {
-    this.pendingStart.forEach(job => {
-      const clearRollbacks = () => {
-        job.rollbacks.length = 0;
-      };
-      job.coroutine = coroutine(job.fn(clearRollbacks, job.abortController.signal), job.rollbacks);
-    });
-    this.pendingStart.length = 0;
-
     this.jobs.forEach((job, eid) => {
+      if (!job.coroutine) {
+        const clearRollbacks = () => {
+          job.rollbacks.length = 0;
+        };
+        job.coroutine = coroutine(job.startCallback(clearRollbacks, job.abortController.signal), job.rollbacks);
+      }
+
       if (job.coroutine!().done) {
-        this.jobs.delete(eid); // TODO Is this safe?
```

Comment:
I used this test to confirm that it was safe to delete from the map during iteration:
```
{
  const map = new Map([
    ["foo", 3],
    ["bar", 4],
    ["xxx", 5],
    ["yyy", 4],
    ["zzz", 9]
  ]);
  map.forEach((v, k) => {
    console.log(v, k);
    if (v === 4) map.delete(k);
  });
  console.log(map);
}
```